scripts/subspace_mcmc_demo.py

```diff
@@ -0,0 +1,175 @@
+'''
+This demo shows the subspace learning procedure using various HMC algorithms and then compares them
+using agreement and total variation distance metrics.
+Author : Aleyna Kara(@karalleyna)
+'''
+
+import superimport
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pickle
+from sklearn.model_selection import train_test_split
+
+import jax
+from jax.random import split
+from jax.experimental import stax
+from jax.experimental.stax import Dense, Relu, LogSoftmax
+import jax.numpy as jnp
+
+from subspace_learning_lib import subspace_learning, loglikelihood, logprior, accuracy
+from subspace_learning_lib import build_sgd, build_nuts_sampler
+import pyprobml_utils as pml
+
+from sgmcmcjax.samplers import *
+
+def load_dataset(filename):
+    with open(filename, "rb") as f:
```

Comment:
rather than using pickle files, I suggest just using
```
from sklearn.datasets import fetch_openml
X, y = fetch_openml('shuttle', version=1, return_X_y=True)
```
Also, removing any reference  to `num_actions`, and pass in the RNG key.